File: src/main/java/com/learn/todolist/infra/UserRepository.java
package com.learn.todolist.infra;

import com.learn.todolist.domain.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {
    Optional<User> findByEmail(String email);
//    Optional<User> findByEmailAndDeletedAtIsNull(String email);
    Boolean existsByEmail(String email);
//    Boolean existsByEmailAndDeletedAtIsNull(String email);
    Boolean existsByNickName(String nickName);
//    Boolean existsByNickNameAndDeletedAtIsNull(String nickName);
}

File: src/main/kotlin/com/todolist/aplication/commonRepository/UserRepository.kt
package com.todolist.aplication.commonRepository

import com.todolist.domain.user.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import java.util.UUID

interface UserRepository: JpaRepository<User, UUID>{
    fun findByEmailAndDeletedAtIsNull(email: String): User?
    fun existsByEmailAndDeletedAtIsNull(email: String): Boolean
    fun existsByNickNameAndDeletedAtIsNull(nickName: String): Boolean
}